### PR TITLE
Remember and restore orientation on fullscreen exit

### PR DESCRIPTION
**Pull request by iampopovich**
- Store the original screen orientation when entering fullscreen.
- Restore the saved orientation when exiting fullscreen.
- On tablets, continue to just toggle the fullscreen UI without changing the device orientation.

#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- record videos
- create clones
- take over the world

#### Before/After Screenshots/Screen Record

- Before:
- After:

#### Fixes the following issue(s)

- Fixes #

#### Relies on the following changes

- 

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [ ] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java [hunk 1]
```diff
@@ -1900,22 +1902,29 @@ public void onFullscreenStateChanged(final boolean fullscreen) {
...
+                : Optional.empty();
+        if (playerUi.isEmpty()) {
             return;
         }
 
-        final int newOrientation = isLandscape
-                ? ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
-                : ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE;
+        // On tablets, just toggle fullscreen UI without orientation change.
+        if (DeviceUtils.isTablet(activity)) {
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java]: TVs should not change the orientation.
```suggestion
        if (DeviceUtils.isTablet(activity) || DeviceUtils.isTv(activity)) {
```

**Review by TobiGr — COMMENTED** [review]
Thank you for the PR. The code LGTM.

I am pretty certain that this code was that simple years ago but was changed to achieve certain behaviour for some edge cases.

We need to test this with landscape and portrait videos (and switching between them) as well as locking the orientation while playing videos on phone, tablet and tv to check for unwanted behaviours.

**Comment by iampopovich** [comment]
it seems to relay to #5276
